src/main/kotlin/org/dif/crypto/JWE.kt

```diff
@@ -56,7 +55,7 @@ fun authEncrypt(payload: String, auth: AuthCryptAlg, from: Key, to: List<Key>):
     val encryptor = when (sender) {
         is ECKey -> ECDH1PUEncrypterMulti(sender, recipients.asKeys())
         is OctetKeyPair -> ECDH1PUX25519EncrypterMulti(sender, recipients.asKeys())
-        else -> throw UnsupportedJWKException(sender.javaClass.name)
+        else -> throw UnsupportedException.JWK(sender.javaClass.name)
```

Comment:
The approach with sealed exceptions (or error codes as enums) is good. 
Not sure about this, but do we really need `UnsupportedException`? Isn't it covered by either `DIDCommInvalidArgumentException` (something is wrong with input parameters) or `MalformedMessageException` (something is wrong with the message to be unpacked)?